### ISSUE TITLE
get_rank for workers

### ISSUE THREAD

**Issue by UnixJunkie** [UnixJunkie/parany#65]
same semantic as parmap would be nice
```ocaml
(** Getting the current worker rank.
    The master process has rank -1. Other processes
    have the rank at which they were forked out (a worker's rank is in [0..ncores-1])  *)
val get_rank : unit -> int
```

**Comment by UnixJunkie** [comment]
I need to know more about the semantic of Domain.self() first.
This does not behave as I expect.

**Comment by UnixJunkie** [comment]
Need to use Domain.DLS.{get|set}

**Comment by UnixJunkie** [comment]
And ```ocaml
Domain.self () :> int
```

**Comment by UnixJunkie** [comment]
functionality available from v13.1.0
